scripts/mc_megaraid_monitor.py

```diff
@@ -0,0 +1,244 @@
+#! /usr/bin/env python
+# -*- mode: python; coding: utf-8 -*-
+# Copyright 2017 the HERA Collaboration
+# Licensed under the 2-clause BSD license.
+
+"""Check in on the status of this machine's MegaRAID array and log information
+to the M&C system.
+
+We assume that there is one controller with one big old RAID array.
+
+This script must be run as root since that's what the storcli64 command line
+client requires.
+
+"""
+from __future__ import absolute_import, division, print_function
+
+import errno
+import json
+import os.path
+import socket
+from subprocess import Popen, PIPE, STDOUT
+import sys
+
+from astropy.time import Time, TimeDelta
+
+from hera_mc import mc
+
+
+# Preliminaries
+#
+# This script runs as root on netboot machines where the majority of the
+# filesystem is readonly. So `event_ticker` is hardcoded to point to somewhere
+# where we can reliably save state.
+#
+# If `num_recent_events` gets too big, the relevant storcli call can start taking
+# incredibly long to complete.
+
+storcli = '/opt/MegaRAID/storcli/storcli64'
+event_ticker = '/home/obs/.hera_mc/megaraid_last_event_id_%s.txt' % (socket.gethostname(), )
+controller = 0
+num_recent_events = 32 # if more than this many events occur between runs, some won't get logged
+hostname = socket.gethostname()
+
+show_all_items = [
+    'Controller Status',
+    'Memory Correctable Errors',
+    'Memory Uncorrectable Errors',
+    'BBU Status',
+    'Physical Drives',
+]
+
+event_header_keep_keys = frozenset(['Code', 'Class', 'Locale', 'Event Description'])
+
+
+_months = {
+    'Jan': 1,
+    'Feb': 2,
+    'Mar': 3,
+    'Apr': 4,
+    'May': 5,
+    'Jun': 6,
+    'Jul': 7,
+    'Aug': 8,
+    'Sep': 9,
+    'Oct': 10,
+    'Nov': 11,
+    'Dec': 12,
+}
+
+def parse_storcli_datetime(text):
+    """Example input text: "Sat May 20 00:16:57 2017". Returns and Astropy Time
+    object.
+
+    TBD: I think these times are in system local time (and not, say, UTC), but
+    it's not clear!
+
+    """
+    dow, month, dom, hhmmss, year = text.split()
+    month = _months[month]
+    dom = int(dom)
+    year = int(year)
+    iso_format = '%04d-%02d-%02d %s' % (year, month, dom, hhmmss)
+    return Time(iso_format, format='iso')
```

Comment:
What time zone gets attached when you call this? The system local time? Or do you need to specify it somehow?